docs/scarpet/api/Auxiliary.md

```diff
@@ -447,38 +515,55 @@ specific data directory is under `world/scripts/foo.data/bar/../baz.nbt`, and sh
 `world/scripts/shared/bar/../baz.nbt`.
 
 You can use app data to save non-vanilla information separately from the world and other scripts.
+Typically you can leave this outside of any function, so it runs as the first thing in the app
 
 Throws `nbt_read_error` if failed to read app data.
 
+<pre>
+global_app_data = parse_nbt(load_app_data());   //Grabs data stored on disk
+if(!global_app_data,                            //If there was no data saved
+    global_app_data = default_value
+);
+</pre>
+
 ### `store_app_data(tag)`
 
 Note:  `store_app_data(tag, file)` and `store_app_data(tag, file, shared?)` usages deprecated. Use `write_file` instead.
 
-Stores the app data associated with the app from the world `/scripts` folder. With the `file` parameter saves 
-immediately and with every call to a specific file defined by the `file`, either in app space, or in the scripts
+Stores the app data associated with the app from the world `/scripts` folder in nbt form. With the `file` parameter
+saves immediately and with every call to a specific file defined by the `file`, either in app space, or in the scripts
 shared space if `shared` is true. Without `file` parameter, it may take up to 10
- seconds for the output file 
+seconds for the output file
 to sync preventing flickering in case this tag changes frequently. It will be synced when server closes.
 
 Returns `true` if the file was saved successfully, `false` otherwise.
 
 Uses the same file structure for exclusive app data, and shared data folder as `load_app_data`.
 
+You might want to put this in the `__on_close()->` function, so it gets called as the last thing, when you want to save
+the app's data. But to do that, your data must be NBT-compatible.
+
+<pre>
+__on_close()->(
+    store_app_data(encode_nbt(global_app_data))
+);
+</pre>
```

Comment:
Actually I think this is automatic, though I don't really remember anymore and would have to check (though I've done similar in the past so I'm not really sure). (non-blocking)